File: project/rest-api/src/main/java/template/ResponseData.java
package template;

import lombok.Getter;

import java.sql.Timestamp;

/**
 * 성공 응답에서 사용하는 템플릿
 */
@Getter
public class ResponseData<T> {
    private static final String SUCCESS_MESSAGE = "성공하였습니다.";
    private static final int SUCCESS_STATUS = 200;
    
    private final Timestamp timestamp;
    private final int status;
    private final String message;
    private final T data;

    public ResponseData(T data) {
        this(new Timestamp(System.currentTimeMillis()), SUCCESS_STATUS, SUCCESS_MESSAGE, data);
    }

    public ResponseData(Timestamp timestamp, int status, String message, T data) {
        this.timestamp = timestamp;
        this.status = status;
        this.message = message;
        this.data = data;
    }
}

File: project/user-interface/rest-api/src/main/kotlin/com/numberbox/ui/restapi/util/response/ResponseData.kt
package com.numberbox.ui.restapi.util.response

import java.sql.Timestamp

/**
 * 응답 템플릿
 */
data class ResponseData<T>(
    val timestamp: Timestamp = Timestamp(System.currentTimeMillis()),
    val status: Int = SUCCESS_STATUS,
    val message: String = SUCCESS_MESSAGE,
    val data: T,
) {
    companion object {
        private const val SUCCESS_MESSAGE = "성공하였습니다."
        private const val SUCCESS_STATUS = 200
    }
}
